framework/src/SPIDevObj.cpp

```diff
@@ -256,6 +385,14 @@ int SPIDevObj::setBusFrequency(DevHandle &h, SPI_FREQUENCY freq_hz)
 
 int SPIDevObj::_setBusFrequency(SPI_FREQUENCY freq_hz)
 {
+#ifdef __RPI2
+    /* implement sensor interface via rpi2 spi */
+    int32_t samplerate_div = (int32_t)((SPI_FREQUENCY_20MHZ / freq_hz) + 0.5f) - 1;
+#define MPUREG_SMPLRT_DIV 0x19
+    DF_LOG_DEBUG("samplereate_div = %d", samplerate_div);
+    return _writeReg(MPUREG_SMPLRT_DIV, samplerate_div);
```

Comment:
I think this is wrong. We want to change the bus frequency of the SPI bus itself. This has nothing to do with the sensor and its samplerate as such.